File: cache/cache-annotation-processor/src/test/java/ru/tinkoff/kora/cache/annotation/processor/testcache/DummyCacheManager.java
package ru.tinkoff.kora.cache.annotation.processor.testcache;

import ru.tinkoff.kora.cache.CacheManager;

import javax.annotation.Nonnull;
import java.util.HashMap;
import java.util.Map;

public class DummyCacheManager<K, V> implements CacheManager<K, V> {

    private final Map<String, DummyCache<K, V>> cacheMap = new HashMap<>();

    @Override
    public DummyCache<K, V> getCache(@Nonnull String name) {
        return cacheMap.computeIfAbsent(name, k -> new DummyCache<>(name));
    }

    public void reset() {
        cacheMap.values().forEach(DummyCache::reset);
    }
}

File: cache/cache-symbol-processor/src/test/kotlin/ru/tinkoff/kora/cache/symbol/processor/testcache/DummyCacheManager.kt
package ru.tinkoff.kora.cache.symbol.processor.testcache

class DummyCacheManager<K, V> {

    private val cacheMap = HashMap<String, DummyCache<K, V>>()

    override fun getCache(name: String): DummyCache<K, V> {
        return cacheMap.computeIfAbsent(name) { DummyCache(name) }
    }

    fun reset() {
        cacheMap.values.forEach { c -> c.reset() }
    }
}
